### PR TITLE
[11.0] Allow cancellation of stock.move when applying push rules

### PR DESCRIPTION
**Pull request by grindtildeath**
TODO :
 - [x] Add test

### CODE REVIEW

**Review by max3903 — APPROVED** [review]

File: stock_picking_restrict_cancel_with_orig_move/tests/test_restrict_cancel.py [hunk 1]
```diff
@@ -51,3 +51,28 @@ def test_restrict(self):
...
+                'product_uom': self.env.ref('product.product_uom_unit').id,
+                'product_uom_qty': 3,
+            })]
+        })
+        qc_to_stock_move = self.env['stock.move'].search([
+            ('product_id', '=', self.dummy_product.id),
+            ('location_id', '=', self.qc_loc.id),
+            ('location_dest_id', '=', self.stock_loc.id),
+        ])
+        # qc_to_stock_move has merged all the moves so it's quantity is 5
```
Comment (simahawk) [inline stock_picking_restrict_cancel_with_orig_move/tests/test_restrict_cancel.py]: *its

File: stock_picking_restrict_cancel_with_orig_move/tests/test_restrict_cancel.py [hunk 2]
```diff
@@ -51,3 +51,28 @@ def test_restrict(self):
         self.input_to_qc_picking.action_cancel()
         self.assertEqual(qc_to_stock_move.state, 'cancel')
         self.assertEqual(self.input_to_qc_picking.move_lines.state, 'cancel')
+
+    def test_do_not_restrict(self):
+        self.env['stock.picking'].create({
```
Comment (simahawk) [inline stock_picking_restrict_cancel_with_orig_move/tests/test_restrict_cancel.py]: a comment on the pre-conditions on why this should work as tested would be nice :)

**Review by simahawk — CHANGES_REQUESTED** [review]
small - not blocking - request: put yourself in out-of-context-ppl shoes and it your future shoes within 1 year and drop a small explanation :wink:

**Comment by max3903** [comment]
@simahawk Can you update your review please?

**Review by simahawk — APPROVED** [review]

**Pull request merged**